### PR TITLE
chore: Introduce PoRepConfig::new_groth16()

### PR DESCRIPTION
**Pull request by hanbu97**
[https://github.com/filecoin-project/rust-fil-proofs/issues/1632](url)

### CODE REVIEW

**Comment by hanbu97** [comment]
https://github.com/filecoin-project/rust-fil-proofs/issues/1631
finished

**Comment by shawnrader** [comment]
> Please rerun the workflow, clippy is fixed. Thank you.

I am still showing some clippy issues after re-running the workflow (some unused imports): https://app.circleci.com/pipelines/github/filecoin-project/rust-fil-proofs/4397/workflows/fd19d319-f5cf-422e-9885-8ff4b8314310/jobs/96937?invite=true#step-105-355

**Comment by hanbu97** [comment]
Got it. Clippy in tests fixed.

File: fil-proofs-tooling/src/bin/benchy/prodbench.rs
Comment (DrPeterVanNostrand) [inline fil-proofs-tooling/src/bin/benchy/prodbench.rs:169]: I think that it still makes sense to allow `benchy` to configure the number of PoRep challenges, as `benchy` is used to benchmark varrying PoRep configurations, not just the the default/production configurations.

File: filecoin-proofs/src/constants.rs
Comment (DrPeterVanNostrand) [inline filecoin-proofs/src/constants.rs:52]: I think that a better name for this function would be `from_sector_size`, however to allow for varying PoRep configurations during benchmarking (e.g. `benchy`), I think that it's better to leave this as a mutable `HashMap` rather than hardcoded constants.

File: filecoin-proofs/src/types/porep_config.rs
Comment (DrPeterVanNostrand) [inline filecoin-proofs/src/types/porep_config.rs:59]: Until additional proof systems are supported in `rust-fil-proofs`, I think that it's better to exclude the "groth16` identifier from the function name as Groth16 is the only proof system currently supported. If/when additional proof systems are supported, I definitely think that it makes sense to use the function name `new_groth16`, but for now, I think that `new` is a better name.

**Review by DrPeterVanNostrand — CHANGES_REQUESTED** [review]
Thank you!

I definitely like the idea of having a simple `PorepConfig` constructor.

My only concerns with this PR are:
1. `PorepConfig::new_groth16` should be named `new` until additional proof systems are supported.
2. Minimum PoRep challenges should be kept as a mutable setting to allow `benchy` to benchmark non-default PoRep configurations.

**Comment by vmx** [comment]
>     1. `PorepConfig::new_groth16` should be named `new` until additional proof systems are supported.

I proposed in the original issue at https://github.com/filecoin-project/rust-fil-proofs/issues/1632 naming it `new_groth16()`. I still think it's a good idea (and less hassle for the future), but if @DrPeterVanNostrand feels strongly about it, we can also change it to just `new()`.

**Review by storojs72 — DISMISSED** [review]
I support this change (though it still requires to fix merge conflicts), as it removes more lines of code than adds (+88; -218), that makes codebase a bit more readable.

**Comment by hanbu97** [comment]
What should I do next, or some tasks else.

**Comment by vmx** [comment]
@hanbu97 The comment about making things again a mutable HashMap is something you could work on. I'll talk with @DrPeterVanNostrand directly whether it should be `new` or `new_groth16` and let you know once we've come to a decision.

**Review by vmx — APPROVED** [review]
Thanks a lot, I really like the changes!

File: filecoin-proofs/src/constants.rs
Comment (DrPeterVanNostrand) [inline filecoin-proofs/src/constants.rs:76]: To me, the name of this function implies that we are deriving the partition count from the number of porep challenges. I think that a better name for this function would be something like `from_sector_size` or `for_sector_size` as we are retrieving the minimum challenge count for a given sector size.

Comment (hanbu97) [reply filecoin-proofs/src/constants.rs]: I agree with you, will make it more readable. Changes done.

**Review by vmx — APPROVED** [review]
Looks good to me. I'm just waiting for @DrPeterVanNostrand for final approval.

File: filecoin-proofs/src/constants.rs
Comment (DrPeterVanNostrand) [reply filecoin-proofs/src/constants.rs]: Awesome, thank you!

**Comment by DrPeterVanNostrand** [comment]
Looks good to me. Thanks @hanbu97!

**Review by vmx — APPROVED** [review]
As @DrPeterVanNostrand also approved it, I hereby approve it.

@hanbu97 could you please rebase one more time? Best would be if you would squash it into a single commit.

**Review by DrPeterVanNostrand — APPROVED** [review]

**Comment by hanbu97** [comment]
> @hanbu97 could you please rebase one more time? Best would be if you would squash it into a single commit.

When I tried to squash commits. Some error occurs.
> Unable to squash. Squashing replays all commits up to the last one required for the squash. A merge commit cannot exist among those commits.

It seems because I merged some changes at developing time. I will pay attention to it next time. If the squash is required I may create a new branch.

**Comment by vmx** [comment]
@hanbu97 I dared to squash it into a single commit for you. Please let me know if you agree with that commit and the commit message.

I did a `git rebase -i master` and the fixed up the commits.

**Review by shawnrader — APPROVED** [review]

**Comment by hanbu97** [comment]
Sure go ahead. Let's make it done. Thank you.

**Review by vmx — APPROVED** [review]

**Comment by vmx** [comment]
Thanks again @hanbu97 and sorry that it took so long to get it merged.

**Pull request merged**
